src/controllers/post.controller.js

```diff
@@ -34,22 +34,17 @@ const updatePost = asyncHandler(async (req, res, next) => {
     throw new ApiError(400, "Incomplete Data");
   }
 
-  const existingPost = await Post.findOne({ _id: postId });
+  const existingPost = await Post.findOne({
+    _id: postId,
+    userId: userIdFromToken,
+  });
   if (!existingPost) {
     throw new ApiError(500, "Post not found");
   }
 
-  if (existingPost.userId.toString() != userIdFromToken) {
-    throw new ApiError(403, "Unauthorized");
-  }
+  existingPost.caption = caption;
+  const updatedPost = await existingPost.save();
```

Comment:
no need for new variable , but it doesn't affect much u can keep it , just a suggestion !